WordPress/Classes/Utility/Migrator/CoreDataIterativeMigrator.swift

```diff
@@ -203,35 +238,33 @@ private extension CoreDataIterativeMigrator {
                                       destinationType: storeType,
                                       destinationOptions: nil)
         } catch {
-            return false
+            throw error
         }
 
         do {
             let backupURL = try makeBackup(at: url)
             try copyMigratedOverOriginal(from: tempDestinationURL, to: url)
             try deleteBackupCopies(at: backupURL)
         } catch {
-            return false
+            throw error
         }
-
-        return true
     }
 
     static func metadataForPersistentStore(storeType: String, at url: URL) throws -> [String: Any]? {
-
-        guard let sourceMetadata = try? NSPersistentStoreCoordinator.metadataForPersistentStore(ofType: storeType, at: url, options: nil) else {
-            let description = "Failed to find source metadata for store: \(url)"
-            throw NSError(domain: "IterativeMigrator", code: 102, userInfo: [NSLocalizedDescriptionKey: description])
+        do {
+            return try NSPersistentStoreCoordinator.metadataForPersistentStore(ofType: storeType, at: url, options: nil)
+        } catch {
+            let originalDescription: String = (error as NSError).userInfo[NSLocalizedDescriptionKey] as? String ?? ""
+            let description = "Failed to find source metadata for store: \(url). Original Description: \(originalDescription)"
+            throw CoreDataIterativeMigrator.error(with: IterativeMigratorErrorCodes.noMetadataForStore, description: description)
```

Comment:
how would this error propagate to the caller? I believe the one from `NSPersistentStoreCoordinator.metadataForPersistentStore` would prevail if an error is thrown.

Reply:
I'm not sure I follow.  🤔

Reply:
I believe that, since we are doing `return try...` , in case `NSPersistentStoreCoordinator.metadataForPersistentStore` throws an error, the `catch` block will be executed, but the error thrown in that block will not be thrown by our `static func metadataForPersistentStore(storeType: String, at url: URL)`.

Reply:
@Gio2018 I would expect that the return wouldn't succeed and the execution will then follow to the `catch` block.

But I'm not sure, is something that we should double-check.

Reply:
Right. If the `try` throws an error execution is interrupted before the `return` is evaluated and instead execution switches to the `catch` block.  Happy to split the logic of that line up so the return is part of a separate statement if it will clarify things!

Reply:
I just did some very quick tests on a playground: if the function throws, looks like the error of the throwing function will be thrown.. So regardless the return, this might be a common problem for any error thrown in a catch block of a throwing function (sounds like a riddle :) ). I'll do some more tests and share with you later

Reply:
If I understand correctly, it sounds like if there is a method `bar` that throws, and its called by a method `baz` via a `do/try/catch`, and `baz` throws inside its catch block, that error thrown by `baz` would actually be the error thrown by `bar`?  Did I understand that right? 
I did a test in a playground with the following code block and it seems that `baz` throws its own error, not the error that was thrown by `bar`.  Did I misunderstand something? 

```
import UIKit
import Foundation

class Foo {

    static func bar() throws -> Bool {
        throw NSError(domain: "bar", code: 1, userInfo: nil)
    }

    static func baz() throws -> Bool {
        do {
            return try bar()
        } catch {
            throw NSError(domain: "baz", code: 2, userInfo: nil)
        }
    }

}

do {
    try Foo.baz()
} catch {
    let err = error as NSError
    print( err.domain )
}

```

Reply:
Yes you are right. I thought I was picturing something, but I was pointing to the wrong direction. On the other hand, there still might be a problem (which predates this PR) in the caller, which is `iterativeMigrate`. The following code in a playground:

`import Foundation

enum FakeError: Error {
    case first
    case second
    case third

    var localizedDescription: String {
        switch self {
        case .first:
            return "FIRST"
        case .second:
            return "SECOND"
        case .third:
            return "THIRD"
        }
    }
}

// Equivalent of NSPersistentStoreCoordinator.metadataForPersistentStore
func firstThrowingFunction() throws -> String {
    print("throwing FIRST Error")
    throw FakeError.first
}

// Equivalent of metadataForPersistentStore
func secondThrowingFunction() throws -> String? {

    do {
        return try firstThrowingFunction()
    } catch {
        print("throwing SECOND Error")
        throw FakeError.second
    }
}

// Equivalent of iterativeMigrate
func thirdThrowingFunction() throws {

    guard let _ = try secondThrowingFunction() else {
        print("throwing THIRD Error")
        throw FakeError.third
    }
}


func test() {

    do {
        try thirdThrowingFunction()
    } catch {
        if let error = error as? FakeError {
            print("THE ERROR IS: " + error.localizedDescription) // prints THE ERROR IS: SECOND
        }
    }
}

test()`

`thirdThrowingFunction` (which is the "equivalent" of our `iterativeMigrate` would actually throw the second, not the third error. Does this make sense?

Reply:
Yep. Makes total sense because there's no do/catch around the try. However, I think this might actually be what we want in `iterativeMigrate`.  The call to `metadataForPersistentStore` can throw, and we want that error to surface.  But if it doesn't throw, there is still a chance that it could return a nil value, at least the way its coded I'm not sure if that's the case in practice. It seemed to me that `iterativeMigrate`'s `throw` statement was intended to handle a cases where nil was returned from `metadataForPersistentStore` rather than try to handle the errors it throws.  Does that make sense?  Am I off base with my read of the code?

Reply:
Yep I think it does make sense and it's pretty clear now, sorry that this took so long.

Reply:
Not at all! Thanks for being so thoughtful and thorough :)